python/activator/raw.py

```diff
@@ -149,36 +157,142 @@ def is_path_consistent(oid: str, visit: FannedOutVisit) -> bool:
+    group : `str`
         The group id from the visit, associating the snaps making up the visit.
-    detector: `int`
+    detector : `int`
         The integer detector id for the image being sought.
-    snap: `int`
+    snap : `int`
         The snap number within the group for the visit.
 
     Returns
     -------
-    prefix: `str` or None
+    prefix : `str` or `None`
         The prefix to a path to the corresponding raw image object.  If it
         can be calculated, then the prefix may be the entire path.  If no
-        prefix can be calculated, None is returned.
+        prefix can be calculated, `None` is returned.
     """
-
-    if instrument not in _LSST_CAMERA_LIST:
+    if microservice:
+        return _query_microservice(microservice=microservice,
+                                   instrument=instrument,
+                                   group=group,
+                                   detector=detector,
+                                   snap=snap,
+                                   )
+    elif instrument not in _LSST_CAMERA_LIST:
         return f"{instrument}/{detector}/{group}/{snap}/"
-    # TODO DM-39022: use a microservice to determine paths for LSST cameras.
     return None
 
 
+def _query_microservice(
+    microservice: str, instrument: str, group: str, detector: int, snap: int
+) -> str | None:
+    """Look up a raw image's location from the raw image microservice.
+
+    Parameters
+    ----------
+    microservice : `str`
+        The URI of the microservice to query.
+    instrument : `str`
+        The name of the instrument taking the image.
+    group : `str`
+        The group id from the visit, associating the snaps making up the visit.
+    detector : `int`
+        The integer detector id for the image being sought.
+    snap : `int`
+        The snap number within the group for the visit.
+
+    Returns
+    -------
+    key : `str` or `None`
+        The raw's object key within its bucket, or `None` if no image was found.
+
+    Raises
+    ------
+    RuntimeError
+        Raised if this function could not connect to the microservice, or if the
+        microservice encountered an error.
+    """
+    detector_name = _DETECTOR_FROM_INT[instrument][detector]
+    uri = f"{microservice}/{instrument}/{group}/{snap}/{detector_name}"
+    try:
+        response = requests.get(uri, timeout=1.0)
+        response.raise_for_status()
+        unpacked = response.json()
+    except requests.Timeout as e:
+        raise RuntimeError("Timed out connecting to raw microservice.") from e
+    except requests.RequestException as e:
+        raise RuntimeError("Could not query raw microservice.") from e
+
+    if unpacked["error"]:
+        raise RuntimeError("Raw microservice had an unidentified error.")
+    if unpacked["present"]:
+        # Need to return just the key, without the bucket
+        path = urllib.parse.urlparse(unpacked["uri"], allow_fragments=False).path
+        # TODO: validate output?
```

Comment:
I don't think you can (should have to) know the structure of the returned URIs.  There's no useful validation you can do.